src/sagas/assignments.js

```diff
@@ -0,0 +1,71 @@
+import {all, call, put, select, takeEvery} from 'redux-saga/effects';
+
+import {
+  getCourses,
+  createClassroomAssignment,
+} from '../clients/googleClassroom';
+import {
+  createProjectSnapshot,
+} from '../clients/firebase';
+import {
+  assignmentCreated,
+  assignmentNotCreated,
+} from '../actions/assignments';
+import {
+  closeAssignmentCreator,
+  updateCourses,
+} from '../actions/ui';
+import {
+  getCurrentProject,
+} from '../selectors';
+import {generateTextPreview} from '../util/compileProject';
+import {createSnapshotUrl} from '../util/exportUrls';
+
+export function* openAssignmentCreator() {
+  const courses = yield call(getCourses);
+  yield put(updateCourses(courses));
+}
+
+export function* assignAssignment(
+  {payload: {selectedCourse, dateInput}}) {
+  yield* createAssignment(
+    selectedCourse,
+    dateInput,
+    'PUBLISHED',
+  );
+}
+
+export function* draftAssignment(
+  {payload: {selectedCourse, dateInput}}) {
+  yield* createAssignment(selectedCourse, dateInput, 'DRAFT');
+}
+
+function* createAssignment(selectedCourse, dateInput, assignmentState) {
+  const project = yield select(getCurrentProject);
+  const snapshotKey = yield call(createProjectSnapshot, project);
+  const assignmentData = {
+    workType: 'ASSIGNMENT',
```

Comment:
Seems like the knowledge of Google Classroom’s API is leaking out a bit here? I would expect the `googleClassroom` client to encapsulate things like `workType` and just provide an interface where we can pass in the basic facts of the assignment from the application’s perspective.